### DISCUSSION TITLE
Error in domoticz-solaredge-modbustcp-plugin/plugin.py

### DISCUSSION THREAD

**Discussion by JohnBons** [addiejanssen/domoticz-solaredge-modbustcp-plugin#46] [Q&A]
Running into an issue on my raspberry domoticz environment:
```
2022-10-24 18:13:51.079 Error: SolarEdgeMod: ----> Line 582 in '/home/bonsj/domoticz/plugins/domoticz-solaredge-modbustcp-plugin/plugin.py'
2022-10-24 18:13:51.079 Error: SolarEdgeMod: ----> Line 200 in '/home/bonsj/domoticz/plugins/domoticz-solaredge-modbustcp-plugin/plugin.py'
2022-10-24 18:14:11.078 Error: SolarEdgeMod: 'onHeartbeat' failed 'AttributeError':''NoneType' object has no attribute 'connected''.
2022-10-24 18:14:11.078 Error: SolarEdgeMod: Exception traceback:
2022-10-24 18:14:11.078 Error: SolarEdgeMod: ----> Line 582 in '/home/bonsj/domoticz/plugins/domoticz-solaredge-modbustcp-plugin/plugin.py'
2022-10-24 18:14:11.078 Error: SolarEdgeMod: ----> Line 200 in '/home/bonsj/domoticz/plugins/domoticz-solaredge-modbustcp-plugin/plugin.py'
```
Modbus itself is working (Downloaded repo and used example.py for testing
```
@raspberrypi:/solaredge_modbus $ ./example.py 192.168.2.7 1502 --json |grep device
    "c_deviceaddress": 1,
```

Environment setup:
Version: 2022.1
Build Hash: c9526851b
Compile Date: 2022-01-31 09:34:32
dzVents Version: 3.1.8
Python Version: 3.9.2 (default, Mar 12 2021, 04:06:34) [GCC 10.2.1 20210110]

pip list 
solaredge-modbus    0.7.2 (updated a few days ago)

can someone help me out?

**Comment by addiejanssen** [comment]
Which version of the plugin are you using?

**Reply by JohnBons** [reply]
i used the pre release domoticz-solaredge-modbustcp-plugin.2.0.1

**Reply by JohnBons** [reply]
domoticz-solaredge-modbustcp-plugin.1.1.1 will make domoticz crash/exit
`domoticz.service - LSB: Home Automation System
     Loaded: loaded (/etc/init.d/domoticz.sh; generated)
     Active: active (exited) since Mon 2022-10-24 18:35:19 CEST; 43s ago`
So i tried the pre release domoticz-solaredge-modbustcp-plugin.2.0.1 which resulted in the error mentionned

**Reply by addiejanssen** [reply]
You have found a bug in 2.0.1.
I have added it as an issue and will look into it.
No quick fix available right now though....

**Reply by addiejanssen** [reply]
Hi @JohnBons,

I have just released version 2.0.2 that should fix your issue.
If it doesn't: can you share the full json output of the example.py?
That helps in understanding what is going on.

Enjoy,

Addie

**Comment by JohnBons** [comment]
Hi Addie,

Sure no problem.

This is the logging on the rasp.
```
raspberrypi:~/domoticz/plugins/domoticz-solaredge-modbustcp-plugin $ git branch -a
* meters

 2022-10-31 22:25:56.772 Error: SolarEdgeMod: 'onStart' failed 'ValueError':'invalid literal for int() with base 10: 'null''.
2022-10-31 22:25:56.772 Error: SolarEdgeMod: Exception traceback:
2022-10-31 22:25:56.772 Error: SolarEdgeMod: ----> Line 598 in '/home/bonsj/domoticz/plugins/domoticz-solaredge-modbustcp-plugin/plugin.py'
2022-10-31 22:25:56.772 Error: SolarEdgeMod: ----> Line 164 in '/home/bonsj/domoticz/plugins/domoticz-solaredge-modbustcp-plugin/plugin.py' 
```

The json output seems ok:
`{
    "c_id": "SunS",
    "c_did": 1,
    "c_length": 65,
    "c_manufacturer": "SolarEdge",
    "c_model": "SE3000H-RW000BNN4",
    "c_version": "0004.0016.0023",
    "c_serialnumber": "740F7780",
    "c_deviceaddress": 1,
    "c_sunspec_did": 101,
    "c_sunspec_length": 50,
    "current": 0,
    "l1_current": 0,
    "l2_current": 0,
    "l3_current": 0,
    "current_scale": -2,
    "l1_voltage": 2362,
    "l2_voltage": 0,
    "l3_voltage": 0,
    "l1n_voltage": 2362,
    "l2n_voltage": 0,
    "l3n_voltage": 0,
    "voltage_scale": -1,
    "power_ac": 0,
    "power_ac_scale": 0,
    "frequency": 49981,
    "frequency_scale": -3,
    "power_apparent": 0,
    "power_apparent_scale": 0,
    "power_reactive": 0,
    "power_reactive_scale": 0,
    "power_factor": 0,
    "power_factor_scale": 0,
    "energy_total": 184029,
    "energy_total_scale": 0,
    "current_dc": 0,
    "current_dc_scale": -32768,
    "voltage_dc": 0,
    "voltage_dc_scale": -1,
    "power_dc": 0,
    "power_dc_scale": 0,
    "temperature": 2570,
    "temperature_scale": -2,
    "status": 2,
    "vendor_status": 0,
    "rrcr_state": 0,
    "active_power_limit": 100,
    "cosphi": 0,
    "commit_power_control_settings": 0,
    "restore_power_control_default_settings": 0,
    "reactive_power_config": 0,
    "reactive_power_response_time": 200,
    "advanced_power_control_enable": 1,
    "export_control_mode": 0,
    "export_control_limit_mode": 0,
    "export_control_site_limit": 0,
    "meters": {},
    "batteries": {}
}`

**Reply by addiejanssen** [reply]
That's interesting. Line 164 is about setting the loglevel in the plugin.
That's related to the log level setting in the Domoticz web interface:

![image](https://user-images.githubusercontent.com/55428983/199288582-18a2df55-6482-4a88-a292-7940b38d812b.png)

Did you select a log level there? If so: can you select another one and save the setting? That should fix the `null` issue.

The json looks okay, although it contains values that I have never seen before (probably a result of a newer version of the solaredge_modbus library that is used by the plugin). But that shouldn't break anything.

**Reply by JohnBons** [reply]
Fixed Removed the plugin
updated domoticz to version Version: 2022.1 (build 14591) 
Added the plugin again
Now i see under devices the extra's from the plugin.

thanks!

**Reply by addiejanssen** [reply]
Well, good to read that the issues are fixed. Enjoy!
